src/Mod/PartDesign/App/ShapeBinder.cpp

```diff
@@ -621,6 +621,20 @@ void SubShapeBinder::update(SubShapeBinder::UpdateOption options) {
                     }
                     if (recomputeCopy && !copied->recomputeFeature(true))
                         copyerror = 2;
+                    if (!copyerror) {
+                        for (auto prop : props) {
+                            if (!App::LinkBaseExtension::isCopyOnChangeProperty(this, *prop))
+                                continue;
+                            auto p = copied->getPropertyByName(prop->getName());
+                            if (p && p->getContainer() == copied
+                                && p->getTypeId() == prop->getTypeId()
+                                && !p->isSame(*prop))
+                            {
+                                std::unique_ptr<App::Property> pcopy(p->Copy());
+                                prop->Paste(*pcopy);
+                            }
+                        }
+                    }

```

Comment:
This code block is basically identical to the block above -- at a minimum it should be refactored so we don't repeat ourselves.

Reply:
Could you please suggest the right solution for it to be in style with the rest of the code? Is it enough to get it to lambda and call it from a couple of places? The context is the same in both cases.
As for the duplication, I just didn't want to touch the existing code, thought the patch is 'more minimal' this way :)